### PR TITLE
Visual Recognition v3 node (first release) - third PR.

### PR DESCRIPTION
**Pull request by ylecleach**
could you please manage this PR for me ? thank you very much soheel

### CODE REVIEW

File: services/visual_recognition/v3.js [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        visualRecognition.createClassifier(params, cbdone);
+        break;
+      case 'retrieveClassifiersList':
+        visualRecognition.listClassifiers(params, cbdone);
+        break;
+      case 'retrieveClassifierDetails':
+        visualRecognition.getClassifier(params, cbdone);
+        break;
+      case 'deleteClassifier':
+        visualRecognition.deleteClassifier(params, cbdone);
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:117]: Codacy Issue found: [Expected indentation of 6 space characters but found 8.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l117)

File: services/visual_recognition/v3.js [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+          var err_id = body.images[0].error.error_id
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_id']= err_id;
+          msg.result['error']= err_desc;
+          console.log('Error:', err_desc);
+          msg.payload='see msg.result.error';
+          node.send(msg); 
+        }
+        else {
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:195]: Codacy Issue found: [Closing curly brace does not appear on the same line as the subsequent block.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l195)

File: services/visual_recognition/v3.js [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+          return;
+        }
+        else if (body.images[0].error)
+        {
+          var err_desc = body.images[0].error.description;
+          var err_id = body.images[0].error.error_id
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_id']= err_id;
+          msg.result['error']= err_desc;
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:190]: Codacy Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l190)

File: services/visual_recognition/v3.js [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+          console.log('Error:', err_desc);
+          msg.payload='see msg.result.error';
+          node.send(msg); 
+        }
+        else {
+          msg.result = {};
+          msg.result['all'] = body;
+          msg.payload='see msg.result'; // to remove any Buffer that could remains
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:198]: Codacy Issue found: [Unexpected comment inline with code.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l198)

File: services/visual_recognition/v3.js [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+          msg.result['error_code']= err.code;
+          if (!err.error)
+            msg.result['error']= err.error;
+          console.log('Error:', err.code);
+          node.error('Error code : ' +  err.code);
+          return;
+        }
+        else {
+          msg.result = 'Deleted classifier_id';
+          msg.payload='see msg.result';
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:241]: Codacy Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l241)

File: services/visual_recognition/v3.js [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+  };
+
+  // Utility function that performs the Watson Visual Recognition call. 
+  // the cleanup removes the temp storage, and I am not sure whether 
+  // it should be called here or after alchemy returns and passed
+  // control back to cbdone.
+
+  function performAction(params, feature, cbdone) {
+
+        var visualRecognition = watson.visual_recognition({
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:91]: Codacy Issue found: [Expected indentation of 4 space characters but found 8.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l91)

File: services/visual_recognition/v3.js [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+      }
+
+      var params = {};
+
+      // This is the callback after the call to the watson service.    
+      // Set up as a var within this scope, so it has access to node, msg etc. 
+      // in preparation for the Watson service action       
+      var actionComplete = function(err, body, other) {
+
+        if (err != null && body==null)
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:172]: Codacy Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l172)

File: services/visual_recognition/v3.js [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+        visualRecognition.recognizeText(params, cbdone);
+        break;
+      case 'createClassifier':
+        visualRecognition.createClassifier(params, cbdone);
+        break;
+      case 'retrieveClassifiersList':
+        visualRecognition.listClassifiers(params, cbdone);
+        break;
+      case 'retrieveClassifierDetails':
+        visualRecognition.getClassifier(params, cbdone);
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:114]: Codacy Issue found: [Expected indentation of 6 space characters but found 8.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l114)

File: services/visual_recognition/v3.js [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+  // control back to cbdone.
+
+  function performAction(params, feature, cbdone) {
+
+        var visualRecognition = watson.visual_recognition({
+          api_key: apikey,
+          version: 'v3',
+          version_date: '2016-05-19'
+        });
+    switch(feature)
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:96]: Codacy Issue found: [Opening curly brace does not appear on the same line as controlling statement.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l96)

File: services/visual_recognition/v3.js [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+
+  RED.httpAdmin.get('/watson-visual-recognition/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+  // Utility functions that check for image buffers, urls and stream data in
+ 
+  function imageCheck(data) {
+    return data instanceof Buffer && imageType(data) !== null;
+  };
```
Comment (yrezgui) [inline services/visual_recognition/v3.js:70]: Codacy Issue found: [Unnecessary semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l70)

**Comment by chughts** [comment]
Processing through another fork

**Pull request closed without merging**